lib/friendly_id/finder_methods.rb

```diff
@@ -59,14 +59,47 @@ def potential_primary_key?(id)
     end
 
     def first_by_friendly_id(id)
-      find_by(friendly_id_config.query_field => id.downcase)
+      find_by(friendly_id_config.query_field => parse_friendly_id(id))
+    end
+
+    # Parse the given value to make it suitable for use as a slug according to
+    # your application's rules.
+    #
+    # This method is not intended to be invoked directly; FriendlyId uses it
+    # internally to process a slug into string to use as a finder.
+    #
+    # However, if FriendlyId's default slug parsing doesn't suit your needs,
+    # you can override this method in your model class to control exactly how
+    # slugs are generated.
+    #
+    # ### Example
+    #
+    #     class Person < ActiveRecord::Base
+    #       extend FriendlyId
+    #       friendly_id :name_and_location
+    #
+    #       def name_and_location
+    #         "#{name} from #{location}"
+    #       end
+    #
+    #       # Use default slug, but lower case
+    #       # If `id` is "Jane-Doe" or "JANE-DOE", this finds data by "jane-doe"
+    #       def parse_friendly_id(slug)
+    #         super.downcase
```

Comment:
This is an interesting usage. It could be better to use the passed-in (original) `slug` argument. But maybe someone would like to compose the parsing with some other modules they wrote. 🤷

Reply:
This method is used by the finder, meaning you don't have the original at this point to query, only the slug from the url

Reply:
Yeah, that's what I meant by original: originally received from the user